### PR TITLE
Enable multiple configuration for multi-trajectory

### PR DESCRIPTION
**Pull request by yutakaoka**
Hi,
This is related to the PR I sent in https://github.com/googlecartographer/cartographer_ros/pull/322.
Thanks,

### CODE REVIEW

File: cartographer/mapping/map_builder.cc [hunk 1]
```diff
@@ -99,6 +99,30 @@ int MapBuilder::AddTrajectoryBuilder(
   return trajectory_id;
 }
 
+int MapBuilder::AddTrajectoryBuilder(
```
Comment (SirVer) [inline cartographer/mapping/map_builder.cc]: Remove the original AddTrajectoryBuilder and take the (to be created) TrajectoryBuilderOptions proto here. `CHECK` that the correct oneof is set here (the one that matches the settings in the original `MapBuilderOptions` and therefore the sparse pose graph).

**Comment by SirVer** [comment]
Let's get this design right before we look at the cartographer_ros pull request. 

Right now the API is confusing - MapBuilder takes a trajectory option in the constructor and in AddTrajectoryBuilder - but there most of the struct is ignored.

My suggestion is to remove `trajectory_builder_{2,3}d_options` from the `MapBuilderOptions` proto and create a new proto `TrajectoryBuilderOptions` that consists of a oneof `{ mapping_2d.LocalTrajectoryBuilderOptions, mapping_3d.LocalTrajectoryBuilderOptions }` and take this in `AddTrajectoryBuilder`.

**Comment by yutakaoka** [comment]
Thank you for your advice. I am just trying the `oneof` feature, but it is failing to compile. The `oneof` seems to be available after protobuf-2.6. We can apt-get only protobuf-2.5 with Ubuntu14.04. Is there any advice?

**Comment by SirVer** [comment]
Unfortunate. I think using a proto with 2 fields, but only one will ever be set is the closest approximation.

File: cartographer/mapping/map_builder.cc (hunk 1 above)
Comment (yutakaoka) [inline cartographer/mapping/map_builder.cc]: I created TrajectoryBuilderOptions proto.

**Comment by SirVer** [comment]
lgtm. Thanks for the contribution!

**Pull request merged**